### PR TITLE
fix(#24): Fix feature -run-test-every and some linting

### PR DESCRIPTION
**Pull request by Atem18**
Hi,

I wanted to run this program every hour but even if you set -run-test-every=1h via command-line, it's still does the test every 2 minutes.

I found why it does and so I am proposing that pull request which fixes the problem.

Kevin.

### CODE REVIEW

File: monitor.go [hunk 1]
```diff
@@ -588,7 +588,6 @@ func runMonitor() {
 			promPushMetric.Observe(time.Since(pushStartTime).Seconds())
 
 			log.Infof("Test successful")
-			duration = 2 * time.Minute
```
Comment (josephschorr) [inline monitor.go]: This isn't a sufficient change, as we change the duration in other places. You'll need to set it to this: `duration, err := time.ParseDuration(*testInterval)`

Comment (Atem18) [reply monitor.go]: @josephschorr You are already defining that line here in the main loop so no need to define it again: https://github.com/coreos/registry-monitor/blob/master/monitor.go#L495

You change duration if it fails to pull image : https://github.com/coreos/registry-monitor/blob/master/monitor.go#L537 and if it fails to push image : https://github.com/coreos/registry-monitor/blob/master/monitor.go#L575 , I think to test again pull and push. But why change duration : https://github.com/coreos/registry-monitor/blob/master/monitor.go#L591 defined by user : https://github.com/coreos/registry-monitor/blob/master/monitor.go#L495 if test is ok ?

I tested in my production and removing that line : https://github.com/coreos/registry-monitor/blob/master/monitor.go#L591 as proposed in my pull request make you code works I think as intended.


Thanks in advance for your input and if you could provide me input on my other pull request : https://github.com/coreos/registry-monitor/pull/22, I would be really thankful ! :D

Comment (josephschorr) [reply monitor.go]: Right, but then we need to change it *back* to the user-specified duration once the operation has succeeded, hence the need to add something back here.

File: monitor.go [hunk 2]
```diff
@@ -129,10 +130,10 @@ func newDockerClient() (*docker.Client, error) {
 		return docker.NewClient(dockerHost)
 	}
 
-	cert_path := os.Getenv("DOCKER_CERT_PATH")
-	ca := fmt.Sprintf("%s/ca.pem", cert_path)
-	cert := fmt.Sprintf("%s/cert.pem", cert_path)
-	key := fmt.Sprintf("%s/key.pem", cert_path)
+	certPath := os.Getenv("DOCKER_CERT_PATH")
```
Comment (josephschorr) [inline monitor.go]: Please don't make other changes in the same commit; feel free to submit as a followup PR

File: monitor.go [hunk 3]
```diff
@@ -491,14 +492,20 @@ func main() {
...
 	mainLoop := func() {
-		duration, err := time.ParseDuration(*testInterval)
+		userDuration, err := time.ParseDuration(*testInterval)
 		if err != nil {
 			log.Fatalf("Failed to parse time interval: %v", err)
 		}
 
 		for {
 			if !firstLoop {
+				if status == false {
```
Comment (josephschorr) [inline monitor.go]: Be better to just leave the other assignments as-is, and only update the duration to the user duration where we previously set it to 2 minutes

**Pull request closed without merging**